packages/browser/core/src/core/auth/internal/StitchUserImpl.ts

```diff
@@ -19,25 +19,58 @@ import {
   StitchCredential,
   StitchUserProfileImpl
 } from "mongodb-stitch-core-sdk";
-import StitchUser from "../StitchUser";
-import StitchAuthImpl from "./StitchAuthImpl";
+import StitchRedirectCredential from "../providers/StitchRedirectCredential";
+import StitchUser, { StitchUserCodec } from "../StitchUser";
+import { prepareRedirect, RedirectFragmentFields, RedirectTypes, StitchAuthImpl } from "./StitchAuthImpl";
 
 export default class StitchUserImpl extends CoreStitchUserImpl
   implements StitchUser {
-  private readonly auth: StitchAuthImpl;
 
   public constructor(
     id: string,
     loggedInProviderType: string,
     loggedInProviderName: string,
     profile: StitchUserProfileImpl,
-    auth: StitchAuthImpl
+    private readonly auth: StitchAuthImpl
   ) {
     super(id, loggedInProviderType, loggedInProviderName, profile);
-    this.auth = auth;
   }
 
   public linkWithCredential(credential: StitchCredential): Promise<StitchUser> {
     return this.auth.linkWithCredential(this, credential);
   }
+
+  public linkUserWithRedirect(
+    credential: StitchRedirectCredential
+  ): Promise<void> {
+    this.auth.setRedirectFragment(
+      RedirectFragmentFields.LinkUser,
+      JSON.stringify(new StitchUserCodec(this.auth).encode(this))
+    );
+
+    const { redirectUrl, state } = prepareRedirect(
+      this.auth,
+      credential,
+      RedirectTypes.Login
+    );
+
+    console.log(credential);
+    const link = this.auth.browserAuthRoutes.getAuthProviderLinkRedirectRoute(
+      credential,
+      redirectUrl,
+      state,
+      this.auth.deviceInfo
+    );
+
+    return fetch(
+      new Request(link, {
+        credentials: "include",
+        headers: {
+          Authorization: "Bearer " + this.auth.authInfo.accessToken
```

Comment:
This move into StitchAuthImpl will let you make auth not be public

Reply:
👍